p01/p01b.jq

```diff
@@ -1,8 +1,7 @@
-#!/usr/bin/env jq -s -f
+#!/usr/bin/env jq -f
```

Comment:
$nextwin and prevwin always share two elements, so instead of adding them, you can just compare the first/last elements (though adding them is clearer)

The transpose method seems simpler/cleaner. I remember you mentioning trying to decide between the two.

I'm guessing the point of reduce is to try to stream over the data once, and avoid the multiple list creation going on with transpose.

I would feel better about reduce if it were true streaming. But since we are using -s(lurp), it means we're reading the whole file into a list first, right? So we're only saving on some small constant factor for memory use.

Is it even possible to do "true streaming" with reduce? I couldn't figure out how to run reduce without first converting stream to a list (or using slurp which is just doing that implicitly right?)